Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/server/TraceLocationServer.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.eventregistration.events.server
+
+import androidx.annotation.VisibleForTesting
+import dagger.Lazy
+import de.rki.coronawarnapp.eventregistration.events.TRACE_LOCATION_VERSION
+import de.rki.coronawarnapp.eventregistration.events.TraceLocationUserInput
+import de.rki.coronawarnapp.server.protocols.internal.pt.TraceLocationOuterClass
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.seconds
+import retrofit2.HttpException
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TraceLocationServer @Inject constructor(
+    private val api: Lazy<CreateTraceLocationApiV1>
+) {
+
+    suspend fun retrieveSignedTraceLocation(
+        traceLocationUserInput: TraceLocationUserInput
+    ): TraceLocationOuterClass.SignedTraceLocation {
+
+        val traceLocationProto = traceLocationUserInput.toTraceLocationProtoBuf()
+        val response = api.get().createTraceLocation(traceLocationProto)
+
+        if (!response.isSuccessful) throw HttpException(response)
+        if (response.body() == null) {
+            throw IllegalStateException("Response is successful, but body is empty.")
+        }
+
+        val byteStream = response.body()!!.byteStream()
+        val signedTraceLocation = TraceLocationOuterClass.SignedTraceLocation.parseFrom(byteStream)
```

Comment:
ByteStream is an InputStream that should be closed, maybe:


```kotlin
        val signedTraceLocation = response.body()?.byteStream()?.use {
            TraceLocationOuterClass.SignedTraceLocation.parseFrom(it)
        } ?: throw IllegalStateException("Response is successful, but body is empty.")
```

Reply:
Thanks for the code suggestion, however it isn't needed anymore because the `ProtoConverterFactory` is now doing the conversion.